portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +36,57 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  /** Returns all questions posted */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Jason</h1>");
+    Query query = new Query("Questions").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Question> questions = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String name = (String) entity.getProperty("name");
```

Comment:
I think a better name for the actual Question text within the Question object would be something like `Question.text` instead of `Question.question`. I'm not sure where your Question class went in this PR (I don't see it. in my code-review portal). But if it's actually Question.question, I think it should be renamed for clarity.